<a href="https://colab.research.google.com/github/AislingGilder85/Project-2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2 - Part 1
 - Aisling Gilder
 - 06/09/2022

## Data Choice 1

1)Source of data
>For the first dataset, I have selected the White Whine data set from https://archive-beta.ics.uci.edu/ml/datasets/wine+quality. This dataset is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license.

In [8]:
import pandas as pd
url = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vS83MAgVaiUDy_7Tv-WW1Kj1zy59DM-ALhkAJY_K70IGBKmPp9-4cFtWO_h1M82cQ/pub?output=xlsx'
df = pd.read_excel(url)
df.head()

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.0              0.27         0.36            20.7      0.045   
1            6.3              0.30         0.34             1.6      0.049   
2            8.1              0.28         0.40             6.9      0.050   
3            7.2              0.23         0.32             8.5      0.058   
4            7.2              0.23         0.32             8.5      0.058   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 45.0                 170.0   1.0010  3.00       0.45   
1                 14.0                 132.0   0.9940  3.30       0.49   
2                 30.0                  97.0   0.9951  3.26       0.44   
3                 47.0                 186.0   0.9956  3.19       0.40   
4                 47.0                 186.0   0.9956  3.19       0.40   

   alcohol  quality  
0      8.8      6.0  
1      9.5      6.0  
2     10.1      6.0  
3      9.9      6.0  
4      9.9      6.0

2) Brief Description of data
> This data set contains values of varied chemical and physical properties of white variates of the Porteguese "Vinho Verde" wine

3) What is the Target
> The target is the 'alcohol' column. 

4) Is this a classification or regression problem?
> This is a regression problem.

5) How many features?
>There are 11 features

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4898 entries, 0 to 4897
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   fixed acidity         4898 non-null   float64
 1   volatile acidity      4898 non-null   float64
 2   citric acid           4898 non-null   float64
 3   residual sugar        4898 non-null   float64
 4   chlorides             4898 non-null   float64
 5   free sulfur dioxide   4898 non-null   float64
 6   total sulfur dioxide  4898 non-null   float64
 7   density               4898 non-null   float64
 8   pH                    4898 non-null   float64
 9   sulphates             4898 non-null   float64
 10  alcohol               4898 non-null   float64
 11  quality               4898 non-null   float64
dtypes: float64(12)
memory usage: 459.3 KB


6) How many rows of data
>There are 4898 rows in this dataset. There is no missing data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> The range of alchol levels in wine are not super varies due to how it is produced. As such, tuning the model may be more diffuclt that normal.

## Data Choice 2

1)Source of data
> The second choice of dataset from https://archive-beta.ics.uci.edu/ml/datasets/chess+king+rook+vs+king+pawn, and is being used under the This dataset is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license.


In [10]:
url2 = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vSQs2zQJNJv71QFjriKJIsgRkEqE4sV4t1X9ToRT9jmtumcycxW-ndCF1Rpv9mfKg/pub?output=xlsx'
df2 = pd.read_excel(url2)
df2.head()

f f.1 f.2 f.3 f.4 f.5 f.6 f.7 f.8 f.9  ... f.23 f.24 f.25 f.26 f.27 f.28  \
0  f   f   f   f   t   f   f   f   f   f  ...    f    f    f    f    f    f   
1  f   f   f   f   t   f   t   f   f   f  ...    f    f    f    f    f    f   
2  f   f   f   f   f   f   f   f   t   f  ...    f    f    f    f    f    f   
3  f   f   f   f   f   f   f   f   f   f  ...    f    f    f    f    f    f   
4  f   f   f   f   f   f   f   f   f   f  ...    f    f    t    f    f    f   

  t.2 t.3 n.1  won  
0   t   t   n  won  
1   t   t   n  won  
2   t   t   n  won  
3   t   t   n  won  
4   t   t   n  won  

[5 rows x 37 columns]

2) Brief Description of data
> This data is from Chess Endgames that are White King and Rook vs Black King and Pawn on A7. The pawn on a7 is one square away from promotion. It is white's turn to move. Each column represents a different board state at the endgame. The names of the feature columns are located on the companion text file: https://docs.google.com/document/d/e/2PACX-1vS_4x2wVnc_DVlWa7ztIouqy0_VlFubcA8PVC_SF4Ze8reqj9CsS2RPbN-RU1jMqg/pub


3) What is the Target
> The Target is the 'won' column, representing if White is able to win or not.



4) Is this a classification or regression problem?
> This is a binary classification problem



5) How many features?
>There are 36 features

In [11]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3195 entries, 0 to 3194
Data columns (total 37 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   f       3195 non-null   object
 1   f.1     3195 non-null   object
 2   f.2     3195 non-null   object
 3   f.3     3195 non-null   object
 4   f.4     3195 non-null   object
 5   f.5     3195 non-null   object
 6   f.6     3195 non-null   object
 7   f.7     3195 non-null   object
 8   f.8     3195 non-null   object
 9   f.9     3195 non-null   object
 10  f.10    3195 non-null   object
 11  f.11    3195 non-null   object
 12  l       3195 non-null   object
 13  f.12    3195 non-null   object
 14  n       3195 non-null   object
 15  f.13    3195 non-null   object
 16  f.14    3195 non-null   object
 17  t       3195 non-null   object
 18  f.15    3195 non-null   object
 19  f.16    3195 non-null   object
 20  f.17    3195 non-null   object
 21  f.18    3195 non-null   object
 22  f.19    3195 non-null   

6) How many rows of data
>There are 3195 rows and no missing data



7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
> The column names are not defined in the data itself. In the companion text file, the feature list is simpy the abbrevations used in the book "The role of structured induction in expert systems" by Alen David Shapiro. It is avialble as a pdf from the Edinburgh Research Archive, https://era.ed.ac.uk/bitstream/handle/1842/30745/ShapiroAD_1984redux.pdf?sequence=1&isAllowed=y.

>In a addition to the extra research and sourcing required for clear column names, different chess terms, such as skewer, threatening a piece, and check may have to be defined as readers may not have the intermediate level chess knowledge to know what those terms mean. 
